File: Wirtualny Swiat/Piotr Przymus 193736 Wirtualny Swiat/Organizm.cpp
#include "Organizm.h"

Organizm::Organizm(int polX, int polY, Swiat& swiat)
	: polozeniex(polX), polozeniey(polY), swiat(swiat)
{
	wiek = 0;
}

Organizm::Organizm(int polozenieX, int polozenieY, int sila, int inicjatywa, char znak, int wiek, string nazwa, Swiat& swiat)
	: polozeniex(polozenieX), polozeniey(polozenieY), sila(sila), inicjatywa(inicjatywa), znak_gatunku(znak), wiek(wiek), nazwa(nazwa), swiat(swiat)
{}

Organizm::~Organizm()
{
}

void Organizm::SetSila(int s)
{
	sila = s;
}

void Organizm::SetPolozenieX(int x)
{
	polozeniex = x;
}

void Organizm::SetPolozenieY(int y)
{
	polozeniey = y;
}

void Organizm::SetWiek(int w)
{
	wiek = w;
}

int Organizm::GetPolozenieX()
{
	return polozeniex;
}

int Organizm::GetPolozenieY()
{
	return polozeniey;
}

int Organizm::GetSila()
{
	return sila;
}

int Organizm::GetInicjatywa()
{
	return inicjatywa;
}

char Organizm::GetZnak_Gatunku()
{
	return znak_gatunku;
}

int Organizm::GetWiek()
{
	return wiek;
}

string Organizm::GetNazwa()
{
	return nazwa;
}

void Organizm::Postarzej()
{
	wiek++;
}

void Organizm::rysowanie()
{
	swiat.Dopisz(polozeniex, polozeniey, znak_gatunku);
}

File: Wirtualny Swiat/PiotrPrzymus193736JavaWS/Organizm.java
package WSJAVA;

abstract public class Organizm {
    protected int polozeniex, polozeniey, sila, inicjatywa, wiek;
    protected char znak_gatunku;
    protected String nazwa;
    protected Swiat swiat;

    public Organizm(int polX, int polY, Swiat swiat)
    {
        this.polozeniex = polX;
        this.polozeniey = polY;
        this.swiat = swiat;
        wiek = 0;
    }

    public Organizm(int polozenieX, int polozenieY, int sila, int inicjatywa, char znak_gatunku, int wiek, String nazwa, Swiat swiat)
    {
        this.polozeniex = polozenieX;
        this.polozeniey = polozenieY;
        this.sila = sila;
        this.inicjatywa = inicjatywa;
        this.znak_gatunku = znak_gatunku;
        this.wiek = wiek;
        this.nazwa = nazwa;
        this.swiat = swiat;
    }

    abstract public void akcja();
    abstract public void kolizja(Organizm o);
    abstract public void Rozmnazanie(Organizm o);
    abstract public void PostawDziecko(int x, int y);
    public void rysowanie()
    {
        swiat.Dopisz(polozeniex, polozeniey, znak_gatunku);
    }

    public void SetSila(int s)
    {
        sila = s;
    }
    public void SetPolozenieX(int x)
    {
        polozeniex = x;
    }
    public void SetPolozenieY(int y)
    {
        polozeniey = y;
    }
    public void SetWiek(int w)
    {
        wiek = w;
    }

    public int GetPolozenieX()
    {
        return polozeniex;
    }
    public int GetPolozenieY()
    {
        return polozeniey;
    }
    public int GetSila()
    {
        return sila;
    }
    public int GetInicjatywa()
    {
        return inicjatywa;
    }
    public int GetWiek()
    {
        return wiek;
    }
    public char GetZnak_Gatunku()
    {
        return znak_gatunku;
    }
    public String GetNazwa()
    {
        return nazwa;
    }
    public void Postarzej()
    {
        wiek++;
    }
}
